pkg/virt-handler/vm.go

```diff
@@ -225,64 +226,32 @@ func MapPersistentVolumes(vm *v1.VM, restClient cache.Getter, namespace string)
 			obj, err = restClient.Get().Resource("persistentvolumes").Name(pvc.Spec.VolumeName).Do().Get()
 
 			if err != nil {
-				logging.DefaultLogger().Error().Reason(err).Msg("unable to access persistent volume record")
+				logger.Error().Reason(err).Msg("unable to access persistent volume record")
 				return vm, fmt.Errorf("unable to access persistent volume record: %v", err)
 			}
 			pv := obj.(*k8sv1.PersistentVolume)
 
-			if pv.Spec.ISCSI != nil {
-				logging.DefaultLogger().Object(vm).Info().Msg("Mapping iSCSI PVC")
-				newDisk := v1.Disk{}
-
-				newDisk.Type = "network"
-				newDisk.Device = "disk"
-				newDisk.Target = disk.Target
-				newDisk.Driver = new(v1.DiskDriver)
-				newDisk.Driver.Type = "raw"
-				newDisk.Driver.Name = "qemu"
-
-				newDisk.Source.Name = fmt.Sprintf("%s/%d", pv.Spec.ISCSI.IQN, pv.Spec.ISCSI.Lun)
-				newDisk.Source.Protocol = "iscsi"
-
-				hostPort := strings.Split(pv.Spec.ISCSI.TargetPortal, ":")
-				ipAddrs, err := net.LookupIP(hostPort[0])
-				if err != nil || len(ipAddrs) < 1 {
-					logging.DefaultLogger().Error().Reason(err).Msgf("Unable to resolve host '%s'", hostPort[0])
-					return vm, fmt.Errorf("Unable to resolve host '%s': %s", hostPort[0], err)
-				}
-
-				newDisk.Source.Host = &v1.DiskSourceHost{}
-				newDisk.Source.Host.Name = ipAddrs[0].String()
-				if len(hostPort) > 1 {
-					newDisk.Source.Host.Port = hostPort[1]
-				}
-
-				// This iscsi device has auth associated with it.
-				if pv.Spec.ISCSI.SecretRef != nil && pv.Spec.ISCSI.SecretRef.Name != "" {
-					newDisk.Auth = &v1.DiskAuth{
-						Secret: &v1.DiskSecret{
-							Type:  "iscsi",
-							Usage: pv.Spec.ISCSI.SecretRef.Name,
-						},
-					}
-				}
-
-				vmCopy.Spec.Domain.Devices.Disks[idx] = newDisk
-			} else {
-				logging.DefaultLogger().Object(vm).Error().Msg(fmt.Sprintf("Referenced PV %v is backed by an unsupported storage type", pv))
+			logger.Info().Msgf("Mapping iSCSI PVC %s", pv.Name)
+			newDisk, err := mapPVToDisk(&disk, pv)
```

Comment:
Previously these iscsi specific log messages were encapsulated by ```if pv.spec.ISCSI``` 

Now, since that is no longer the case, if we add more PV support outside of ISCSI, all these log messages have to be updated.

Reply:
I think, we can just change that, when we start supporting other PVCs?

Reply:
My concern is it might not be obvious that log message needs to change in the future. This function is generic.

Reply:
Done